From: Gerald Schaefer <geraldsc@de.ibm.com>
Subject: IBM Codestream linux-2.6.16 october2005, patch 05-06

  Description: dasd: dasd_eer returns data in wrong format
  Symptom:     The dasd_eer device returns sense data in a binary record format.
               Each record starts with a fixed set of data fields describing the
               record, followed by a variable set of sense data bytes.
               With this problem however, there were 6 additional bytes of
               random data between the fixed part and the variable lenght part.
  Problem:     The fixed part of the record is defined as a struct in the DASD
               eer code. Since this struct is part of a binary interface it
               should have been declared with __attribute__ ((packed));
               As this was omitted, the compiler of the test system (SLES10 RC2)
               added 6 extra bytes of padding to the struct.
  Solution:    Add __attribute__ ((packed)); to struct dasd_eer_header.
  Problem-ID:  24647
  Archs:       s390-31, s390-64

Acked-by: Hannes Reinecke <hare@suse.de>

--- linux-2.5/drivers/s390/block/dasd_eer.c	16 May 2006 16:22:37 -0000	1.8.2.2
+++ linux-2.5/drivers/s390/block/dasd_eer.c	13 Jun 2006 16:03:27 -0000	1.8.2.3
@@ -276,7 +276,7 @@
 	__u64 tv_sec;
 	__u64 tv_usec;
 	char busid[DASD_EER_BUSID_SIZE];
-};
+} __attribute__ ((packed));
 
 /*
  * The following function can be used for those triggers that have

